kratos/includes/properties.h

```diff
@@ -635,7 +630,36 @@ class Properties : public IndexedObject
 
     ///@}
     ///@name Private Operations
-    ///@{
+    ///@
+
+    /**
+     * @brief Prints the data of an object of type TClass to the given output stream with indentation.
+     * @param rOStream The output stream where the data will be printed.
+     * @param rThisClass The object of type TClass whose data will be printed.
+     * @param Identation (optional) The string used for the indentation. Default is four spaces.
+     */
+    template<class TClass>
+    static void PrintDataWithIdentation(
+        std::ostream& rOStream,
+        const TClass& rThisClass,
+        const std::string Identation = "\t"
+        )
+    {
+        // Auxiliary stream and line
+        std::stringstream ss;
+        std::string line;
+        rThisClass.PrintData(ss);
+
+        // Get the output string from the stringstream.
+        const std::string& r_output = ss.str();
+
+        // Split the output string into lines.
+        std::istringstream iss(r_output);
+        while (std::getline(iss, line)) {
+            // Here, 'line' is a single line from the output.
+            rOStream << Identation << line << "\n";
+        }
```

Comment:
Not 100% sure, but maybe we can place this in the `StringUtilities` as it might be useful for similar purposes in other classes. Even the `PrintDataWithIdentation` maybe... What you think?

Reply:
Okay

Reply:
MOved